### PR TITLE
CMP-1500 getuserstatus implementation

### PR DESCRIPTION
**Pull request by nicolas-chaix-didomi**
Closes: CMP-1500

- Update native SDKs
- implement `getUserStatus` method
- UI tests for `getUserStatus` method
- Fix UI tests for notice

### CODE REVIEW

File: src/Didomi.ts [hunk 1]
```diff
@@ -285,6 +285,12 @@ export const Didomi = {
       vendorId
     ),
 
+  /**
+   *  Get the user consent status.
+   *  @returns: status that represents user consent.
+   */
+  getUserStatus: (): Promise<any> => RNDidomi.getUserStatus(),
```
Comment (pmerlet-at-didomi) [inline src/Didomi.ts:292]: We should also deprecate the other methods, with a message indicating the equivalent using getUserStatus

Comment (pmerlet-at-didomi) [reply src/Didomi.ts:292]: getUserConsentStatusForPurpose, etc.

**Review by pmerlet-at-didomi — DISMISSED** [review]

File: android/src/main/java/com/reactnativedidomi/DidomiModule.kt [hunk 2]
```diff
@@ -70,6 +70,16 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
          */
         override fun noticeClickPrivacyPolicy(event: NoticeClickPrivacyPolicyEvent) = prepareEvent(EventTypes.NOTICE_CLICK_PRIVACY_POLICY.event, null)
 
+        /**
+         * The preferences is being hidden
```
Comment (felipe-saez) [inline android/src/main/java/com/reactnativedidomi/DidomiModule.kt:74]: should this read something like `The preferences screen is being hidden` or `The preferences are being hidden`?

File: android/src/main/java/com/reactnativedidomi/DidomiModule.kt [hunk 3]
```diff
@@ -70,6 +70,16 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
...
+         */
+        override fun hidePreferences(event: HidePreferencesEvent) = prepareEvent(EventTypes.HIDE_PREFERENCES.event, null)
+
+        /**
+         * The preferences is being shown
```
Comment (felipe-saez) [inline android/src/main/java/com/reactnativedidomi/DidomiModule.kt:79]: same as above

File: android/src/main/java/com/reactnativedidomi/DidomiModule.kt (hunk 2 above)
Comment (nicolas-chaix-didomi) [reply android/src/main/java/com/reactnativedidomi/DidomiModule.kt:74]: will do the same for the notice

File: android/src/main/java/com/reactnativedidomi/DidomiModule.kt (hunk 3 above)
Comment (nicolas-chaix-didomi) [reply android/src/main/java/com/reactnativedidomi/DidomiModule.kt:79]: will do the same for the notice

**Review by pmerlet-at-didomi — APPROVED** [review]

**Review by felipe-saez — APPROVED** [review]

**Pull request merged**